File: C#/BluffinMuffin.Poker.DataTypes/IPokerGame.cs
﻿using BluffinMuffin.Poker.DataTypes.EventHandling;

namespace BluffinMuffin.Poker.DataTypes
{
    public interface IPokerGame
    {
        PokerGameObserver Observer { get; }

        TableInfo Table { get; }

        bool PlayMoney(PlayerInfo p, int amnt);
        bool LeaveGame(PlayerInfo p);
        int SitIn(PlayerInfo p, int noSeat = -1, int moneyAmount = 1500);
        bool SitOut(PlayerInfo p);

        string Encode { get; }

        bool IsPlaying { get; }
    }
}

File: JAVA/PokerWorld/src/bluffinmuffin/poker/IPokerGame.java
package bluffinmuffin.poker;

import bluffinmuffin.poker.entities.PlayerInfo;
import bluffinmuffin.poker.entities.TableInfo;
import bluffinmuffin.poker.observer.IPokerGameListener;

public interface IPokerGame {

    /**
     * Debute l'ecoute de tous les evenements de la Game
     * 
     * @param listener
     */
    void attach(IPokerGameListener listener);

    /**
     * Termine l'ecoute de tous les evenements de la Game
     * 
     * @param listener
     */
    void detach(IPokerGameListener listener);

    /**
     * La table associee a la game
     * 
     * @return
     */
    TableInfo getTable();

    /**
     * L'argent jouee par un joueur precis
     * 
     * @param player
     * @param amount
     * @return
     */
    boolean playMoney(PlayerInfo player, int amount);

    /**
     * Un joueur precis quitte la Game
     * 
     * @param player
     * @return
     */
    boolean leaveGame(PlayerInfo player);

    public String encode();
}
